src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -643,6 +635,27 @@ public void RecordAssertion(AssertionStatus status, string message)
             RecordAssertion(status, message, null);
         }
 
+
+        /// <summary>
+        /// Creates a failure message incorporating failures
+        /// from a Multiple Assert block for use by runners
+        /// that don't know about AssertionResults.
+        /// </summary>
+        /// <returns>Message as a string</returns>
+        public string CreateLegacyFailureMessage()
+        {
+            var writer = new StringWriter();
```

Comment:
Isn't this `IDisposable`? I'd strongly suggest wrapping in an `using` block.

Reply:
I mean... maybe. StringWriter is a minimal wrapper around a StringBuilder. So long as the instance doesn't leave this scope this doesn't bother me either way. Of course then, why not use a StringBuilder directly?

Reply:
I know it's not that big of a deal here, but it is a known good practice to always dispose `IDisposable` instances. There is just no reason not to in this case as well.

I don't mind if you want to change it into a `StringBuilder` though.

Reply:
The argument in favor of doing this is that StringWriter may change to do something like lease a StringBuilder and return it to the pool on Dispose. Even though I can imagine the framework compat council vetoing that because people have not always disposed it, you are right about this.